expected/maproom.py

```diff
@@ -88,21 +91,25 @@ def data_tiles(tz, tx, ty, variable, month): # Lesson 14
     Input("variable", "value"),
     Input("month", "value"), # Lesson 14
 )
-def set_colorbar(variable, month): # Lesson 14
+def set_colorbar(variable, this_month): # Lesson 14
     # Lesson 12 starts
-    if variable == "prcp":
+    if variable == "pre":
         data_file = CONFIG["prcp_file"]
         ticks_sample = 10 # Lesson 13
     else:
         data_file = CONFIG["temp_file"] # Lesson 12 ends
         ticks_sample = 1 # Lesson 13
-    data = xr.open_dataarray(
+    data = xr.open_dataset(
         # "data/CRUprcp.nc", Lesson 10 starts
         #DATA_DIR + CONFIG["prcp_file"], # Lesson 10 ends
         DATA_DIR + data_file, # Lesson 12
         decode_times=False
-    ) #.isel(T=-1) Lesson 14
-    data = data.groupby(data["T"] % 12 + 0.5).mean().sel(T=month) # Lesson 14
+    )
+    data["T"].attrs["calendar"] = "360_day"
+    data = xr.decode_cf(data, decode_times=True).rename(
+        {"X": "lon", "Y": "lat"}
+    )[variable] # .isel(T=-1) # Lesson 5, 14
+    data = data.groupby("T.month").mean().sel(month=int(this_month))
```

Comment:
By calculating the mean before selecting the month, you're doing 12x more computation than needed.
```
data.sel(month=int(this_month)).mean()
```

Reply:
Right. I actually made another change that takes us closer to the ideal set up: which is that the colorscale goes from min to max across the months and thus doesn't change with months. this is more appropriate typically for such a Maproom so that months can be compared between them.

An ideal set up would probably have decent default values but then let the user give the limits of their colorscale.